actions/class.ItemContent.php

```diff
@@ -217,37 +181,47 @@ public function download(): void
     /**
      * Delete a file from the item directory
      *
-     * @throws common_exception_MissingParameter
+     * @throws MissingParameterException|TaoMediaException
      */
-    public function delete()
+    public function delete(): void
     {
-        if (!$this->hasRequestParameter('uri') || !$this->hasRequestParameter('path') || !$this->hasRequestParameter('lang')) {
-            throw new common_exception_MissingParameter();
-        }
-
-        $item = $this->getResource($this->getRequestParameter('uri'));
-        $itemLang = $this->getRequestParameter('lang');
+        $params = $this->getRequiredQueryParams('uri', 'lang', 'path');
```

Comment:
https://github.com/oat-sa/extension-tao-item/pull/466/files?diff=split&w=1#r616403830

The same for deletion and other areas changed